lib/Sema/TypeCheckDecl.cpp

```diff
@@ -868,6 +868,12 @@ static void checkRedeclaration(TypeChecker &tc, ValueDecl *current) {
   if (!currentFile || currentDC->isLocalContext())
     return;
 
+  // Cannot define a type called 'Any'
+  if (current->getFullName() == tc.Context.Id_Any) {
```

Comment:
I don't think we should be doing this via lookup at all; if someone writes `Any`, they should get a TypeExpr right from the start.